### PR TITLE
fix: Make failure highlight more visible

### LINKED ISSUE #5610: [Usability] Explore making failure instance highlight more visible
## Describe the bug
In assessment, we highlight elements to review with a thin red line. In certain cases, it can be hard to understand what elements have been highlighted

**To Reproduce**

1. Go to Accessible university
2. Open Assessment > Images > 14.1
3. Turn on visual helper

## Expected behavior

Explore making the visual highlight clearer in different background colors. Design team suggested trying multiple border colors with red in the center and white surrounding this.
Another option to explore would be making the red border slightly thicker and giving a bit of padding between the element and the line 

The major concern is adding too many colors and lines might cover other elements thus why we are filing this bug to explore these two suggestions and discuss.

## Screenshots

example of multiple border colors with a orange background and red and white borders
image

## Context (please complete the following information)

- OS Name & Version:  N/A
- AI-Web Version & Environment:  Prod 2.32
- Browser Version:  N/A
- Target Page:  N/A 

## Are you willing to submit a PR?
No
## Did you search for similar existing issues?
Yes
## Additional context

usability. Bug to explore alternatives

### PR DESCRIPTION
**Pull request by segamiken**
##### Motivation

close #5610

##### Context

> Design team suggested trying multiple border colors with red in the center and white surrounding this.

I implemented this based on one of the design team's suggestions listed in the issue. 

This will make failure highlight more visible, especially for a dark background as in the following picture. However, for pages with white or other background colors, it is not very effective.

##### screenshots/GIFs to description above

Dark Background
| Before | After |
| ----- | ----- |
|スクリーンショット 2022-11-03 20 00 00|スクリーンショット 2022-11-03 19 58 34|

White Background
| Before | After |
| ----- | ----- |
|スクリーンショット 2022-11-03 20 14 26|スクリーンショット 2022-11-03 20 11 11|

#### Pull request checklist

- [x] Addresses an existing issue: #5610
- [x] Ran `yarn null:autoadd`
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [x] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

**Comment by segamiken** [comment]
@microsoft-github-policy-service agree

**Comment by segamiken** [comment]
@microsoft-github-policy-service agree

**Comment by dbjorge** [comment]
Hi @segamiken ,

Thanks very much for the contribution! I think this is definitely on the right track visually, @ferBonnin and I have discussed and agree that the new UX you've created is a big improvement over the current highlight boxes.

I noticed while testing your changes locally that the new boxes look like they render a pixel or two off, and also that they appear to render the inner white border overtop the inside of the element being highlighted; we think we'd prefer to keep all 3 layers of the highlight box outline on the outside of the element if possible. I've been experimenting locally with an updated version of your changes that use a slightly different CSS technique to achieve that vs what you've done here, with a white box-shadow behind a colored outline instead of using a pseudo-element - it seems promising, but the experiment I have locally needs a bit more work before I send it out.

I just wanted to send a status update and acknowledge your contribution - we appreciate your PR, it's just taking us a bit longer than usual to review!

**Comment by segamiken** [comment]
Hi @dbjorge!
Thank you for sharing current status!

> I noticed while testing your changes locally that the new boxes look like they render a pixel or two off, 

Oh, sorry and thank you for finding this!

> it seems promising, but the experiment I have locally needs a bit more work before I send it out.

I got it!
I'm waiting for you to share it with me when you're done with your experiment!

**Comment by dbjorge** [comment]
I've pushed an update that swaps to the box-shadow + outline version. It updates all the box drawers and their labels. It pushes the determination of box width to CSS to reduce the amount of per-element overhead in `style` attributes. It touches a bunch of files to change some `DrawerConfiguration` types from using `borderColor` to `outlineColor`. I also made some related refactorings while patching up the tests:

* The `HighlightBoxDrawer` previously allowed for no formatter to be passed and contained a "default drawer configuration" for that case. This was never used except for tests; I've removed it and updated `drawer.test.ts` accordingly
* Removed an obsolete `LandmarkValue` type (not used anywhere and referenced `borderColor`)

I also fixed up some issues specific to the landmark visualizer; per discussion with @ferBonnin, I included the same white border there but reduced the width of the dashed part from 3px to 2px. I also fixed an issue I noticed with high-contrast-mode rendering of the landmark visualizer's dashed highlight boxes (previously, we used HC colors for the outer box, but not the text label box; this fixes it to be consistent).

For reviewers, I recommend including the Deque Mars test page among your smoke tests; it has a lot of failures against non-white backgrounds that are good test cases.

Screenshots of new behavior:

ad-hoc automated checks

ad-hoc headings

ad-hoc landmarks

Tab stops is not changed by this update and would need to be addressed in a separate PR: 
tab stops

In Windows HC mode:

automated checks visualizer in Windows high contrast Aquatic

landmarks visualizer in Windows high contrast Aquatic

File: src/injected/styles/injected.scss [hunk 1]
```diff
@@ -255,34 +255,70 @@
...
+            outline-offset: inherit !important;
+            box-shadow: inherit !important;
+        }
+
+        .insights-highlight-outline-solid {
+            // Non-HC outline-color is intentionally unset; individual formatters should override it
+            outline-width: 1px !important;
+            outline-style: solid !important;
+            outline-offset: 1px !important;
+            box-shadow: 0 0 0 3px white !important;
```
Comment (peterdur) [inline src/injected/styles/injected.scss:300]: Opinion: I find 1px outline over 3px shadow makes the red outline not stand out as much as I'm used to, especially around images. 
image
I think 2px outline over 4px shadow is a little clearer:
image
Will totally defer to @ferBonnin here.

Comment (ferBonnin) [reply src/injected/styles/injected.scss:300]: I see what you mean @peterdur specially for white backgrounds, I agree that 2px makes it more clearer but it also uses more space. I don't feel strongly about one way or another so if we prefer it, lets go with 2px with 4px shadow 🙂

**Review by peterdur — APPROVED** [review]
Implementation looks good, built and ran locally. Just one question about whether we want the outline to be a little stronger.

**Comment by dbjorge** [comment]
Thanks again for your contribution, @segamiken ! This change will be included in the next Accessibility Insights for Web release.

**Comment by segamiken** [comment]
@dbjorge 
Thank you for your review and refactoring!

**Pull request merged**